### PR TITLE
enhancement(kubernetes source): Upgrade kube and k8s-openapi crate

### PR DESCRIPTION
**Pull request by MOZGIII**
Upgrade `kube` and `k8s-openapi`.

This is blocked by the HTTP sink upgrade, I'll have to address that first. It's on hold while @LucioFranco is working on unifying the HTTP sink, so I'm doing other stuff in the meantime.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1156,6 +1181,12 @@ dependencies = [
  "futures 0.1.29",
 ]
 
+[[package]]
+name = "futures-timer"
```
Comment (LucioFranco) [inline Cargo.lock]: What crate needs this, I've seen some recent bugs with this crate. As in UB.

Comment (LucioFranco) [reply Cargo.lock]: I just pinged the author of kube, lets see what they say but this shouldn't block this.

**Comment by binarylogic** [comment]
@MOZGIII I believe this is no longer blocked. https://github.com/timberio/vector/pull/2111 is merged, which I assume is what you're referring to.

**Comment by MOZGIII** [comment]
Yes, I'll be able to continue on this. We still need to upgrade `http` sink to `hyper` `0.13` first, but I can start with that work now.

**Comment by binarylogic** [comment]
@LucioFranco @MOZGIII given that Lucio is switching the to Tokio transition project and Mike is switching to the Kubernetes project, who should finish this?

**Comment by MOZGIII** [comment]
This is a part of `tokio` upgrade, so @LucioFranco should take over.
If I see a need to touch these dependencies as part of my k8s work - I'll post a message here and we'll sync the works so that it doesn't overlap. Either that, or I'll create a separate issue.

**Comment by LucioFranco** [comment]
@MOZGIII can we just merge this? Code looks fine to me.

**Comment by MOZGIII** [comment]
It's incomplete and doesn't work (event compile) yet. The problem is that `kube` uses `http::request::Request` type internally, and we have to make sure it's from the new hyper. The problem is that in our current implementation, this `http::request::Request` comes from `http` sink. So we're stuck on `http` sink upgrade here.

I didn't look deep into that though, maybe it's easy to fix or work around.

**Comment by LucioFranco** [comment]
Ok, I will take a look.

**Comment by MOZGIII** [comment]
This is obsolete since #2488.

**Pull request closed without merging**